client.go

```diff
@@ -5,22 +5,11 @@ import (
 	"context"
 	"fmt"
 	"io"
-	"time"
 )
 
-// SerialPort is an interface for a serial used by the Client for communication with an Xsens device.
-type SerialPort interface {
-	io.ReadWriteCloser
-	SetReadDeadline(time.Time) error
-	SetWriteDeadline(time.Time) error
-}
-
-// nolint:lll
-//go:generate go run github.com/golang/mock/mockgen -destination internal/gen/mockxsens/mocks.go -package mockxsens . SerialPort
-
 // Client for communicating with an Xsens device.
 type Client struct {
-	p               SerialPort
+	p               io.ReadWriteCloser
```

Comment:
I guess the name p is a relic from it being a port previously, might update to be more clear.